sync3/caches/global.go

```diff
@@ -53,6 +53,33 @@ type EventData struct {
 	// Flag set when this event should force the room contents to be resent e.g
 	// state res, initial join, etc
 	ForceInitial bool
+
+	// ForcePrevBatch is nonempty when the proxy doesn't know the timeline event immediately
+	// prior to this one. Its value is the prevBatch for this event. If set, this token MUST
+	// be sent to sync3 clients (or else there'll be a gap in their timeline).
+	ForcePrevBatch string
```

Comment:
TODO: do we still need this? Should it be pulled out to another PR?

Reply:
The motivation was: https://github.com/matrix-org/sliding-sync/blob/b8e34a7958d81ed30136bbecf33f368979baa34e/sync3/handler/connstate_live.go#L237-L243

Reply:
So the scenario here is seeing in a live update that you've joined a room via a gappy poll.

I think this should actually be impossible now, because we'll kill your connection.